src/runtime/actor/envelope.rs

```diff
@@ -0,0 +1,50 @@
+// Copyright 2022 IOTA Stiftung
+// SPDX-License-Identifier: Apache-2.0
+
+use std::{fmt::Debug, pin::Pin};
+
+use async_trait::async_trait;
+
+use super::{context::ActorContext, Actor};
+
+#[async_trait]
+pub trait HandleEvent<E: Send + Debug>: Actor + Sized {
+    async fn handle_event(
+        &mut self,
+        cx: &mut ActorContext<Self>,
+        event: E,
+        data: &mut Self::Data,
+    ) -> Result<(), Self::Error>;
+}
+
+pub trait DynEvent<A: Actor>: Debug {
+    fn handle<'c, 'a>(
+        self: Box<Self>,
+        cx: &'c mut ActorContext<A>,
+        act: &'c mut A,
+        data: &'c mut A::Data,
+    ) -> Pin<Box<dyn core::future::Future<Output = Result<(), A::Error>> + Send + 'a>>
+    where
+        Self: 'a,
+        'c: 'a;
```

Comment:
Do we need the `'c` lifetime here, if it's the same as `'a`?

Reply:
Egh. This in particular took an incredible amount of tinkering to get working. Maybe there's a better way to express this but I couldn't find it.

Reply:
When I removed the `'c` it still compiled, so it must be working :trollface:

Reply:
Apparently it's fine. Weird.